DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky/src/cholesky.hpp

```diff
@@ -0,0 +1,161 @@
+#ifndef __CHOLESKY_HPP__
+#define __CHOLESKY_HPP__
+
+#include <CL/sycl.hpp>
+#include <chrono>
+#include <cstring>
+#include <sycl/ext/intel/ac_types/ac_complex.hpp>
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <type_traits>
+#include <vector>
+
+#include "memory_transfers.hpp"
+#include "streaming_cholesky.hpp"
+#include "tuple.hpp"
+
+// Forward declare the kernel and pipe names
+// (This prevents unwanted name mangling in the optimization report.)
+class CholeskyDDRToLocalMem;
+class Cholesky;
+class CholeskyLocalMemToDDRL;
+class APipe;
+class QPipe;
+class RPipe;
+
+/*
+  Implementation of the Cholesky decomposition using multiple streaming kernels
+  Can be configured by datatype, matrix size (must use square matrices), real
```

Comment:
If it must be square, why are we specifying rows and columns separately?  And even if there is a good reason for that, shouldn't we check that rows==columns with a static assert?

Reply:
I've modified the code to only take a "dimension" parameter rather than row and column parameters